lib/Parser/Parser.cpp

```diff
@@ -847,7 +868,23 @@ bool Parser::parseLine(std::string &ErrStr) {
       Block *B = 0;
 
       if (IK == Inst::Var) {
-        Inst *I = IC.createVar(InstWidth, InstName);
+        llvm::APInt Zero(InstWidth, 0, false), One(InstWidth, 0, false),
+                    ConstOne(InstWidth, 1, false);
+        if (CurTok.K == Token::KnownBits) {
+          if (InstWidth != CurTok.PatternString.length()) {
+            ErrStr = makeErrStr(TP, "knownbits pattern must be of same length as var width");
+            return false;
+          }
+          for (unsigned i=0; i<InstWidth; ++i) {
+            if (CurTok.PatternString[i] == '0')
+              Zero += (ConstOne.shl(CurTok.PatternString.length()-1-i));
+            else if (CurTok.PatternString[i] == '1')
+              One += (ConstOne.shl(CurTok.PatternString.length()-1-i));
```

Comment:
I'm not sure the parens around the right-hand argument to += are helpful. Can you explain the purpose?